MapboxCoreNavigation/RouteController.swift

```diff
@@ -315,6 +335,40 @@ open class RouteController: NSObject {
     }
     
     /**
+     Checks whether a new location is quality by comparing it to the previous location.
+     
+     We're attempting to throw out coordinates that have a large difference in location or time.
+     */
+    func isQualityLocation(location: CLLocation?) -> Bool {
+        
+        // Any time we return in this func, reset the values.
+        defer {
+            if let location = location {
+                if let lastLocation = previousKnownLocation {
+                    previousKnownLocation?.locationDelta = location.distance(from: lastLocation.location)
+                    previousKnownLocation?.durationDelta = location.timestamp.timeIntervalSince(lastLocation.location.timestamp)
+                }
+                
+                previousKnownLocation?.location = location
+            }
+        }
+        
+        guard let lastLocation = previousKnownLocation, let location = rawLocation else {
+                return false
+        }
+        
+        guard location.distance(from: lastLocation.location) <= lastLocation.locationDelta * 3 else {
+            return false
+        }
+        
+        guard location.timestamp.timeIntervalSince(lastLocation.location.timestamp) <= lastLocation.durationDelta * 3 else {
```

Comment:
If the user enters a tunnel and doesn’t exit for over 3 seconds, does the next location get tossed out as a poor-quality location?